libraries/Microsoft.Bot.Builder.Dialogs/Choices/ChoiceFactory.cs

```diff
@@ -66,22 +67,26 @@ public static Activity Inline(IList<Choice> choices, string text = null, string
 
             // Format list of choices
             var connector = string.Empty;
-            var txt = text ?? string.Empty;
-            txt += " ";
-
+            var txtBuilder = new StringBuilder()
```

Comment:
I would actually suggest passing `text` to the constructor of the `StringBuilder` to let it initialize itself with that. This will make sure to "right size" the `StringBuilder` from the get-go and also eliminate the overhead of the immediate `.Append(text)` call which might have to grow the internal buffer if the `text` is greater than the default capacity (which is `16` chars).